src/models/hooks/goal.js

```diff
@@ -57,6 +58,22 @@ const preventNameChangeWhenOnApprovedAR = (sequelize, instance) => {
   }
 };
 
+const invalidateSimilarityScores = async (sequelize, instance, options) => {
+  const changed = Array.from(instance.changed());
+
+  if (changed.includes('name')) {
+    await sequelize.models.SimScoreCache.destroy({
+      where: {
+        [Op.or]: [
+          { goal1: instance.id },
+          { goal2: instance.id },
+        ],
+      },
+      transaction: options.transaction,
+    });
```

Comment:
is destroy the right path, could we just set the score to null? Since we know the record will be needed again soon

Reply:
I really don't think there's much difference, but I could be wrong. Is there a good reason to make this change that I'm not seeing?